### PR TITLE
Dhcp fixedaddress extend

### PR DESCRIPTION
**Pull request by ExaneServerTeam**
Add a few fields for fixedaddress object class
+ name
+ template

### CODE REVIEW

**Comment by ExaneServerTeam** [comment]
@adsri 
Hello could you please have a look at this merge request ?

File: infoblox_client/objects.py [hunk 1]
```diff
@@ -779,12 +779,12 @@ def ip(self, ip):
...
     _infoblox_type = 'fixedaddress'
-    _fields = ['ipv4addr', 'mac', 'network_view', 'extattrs', 'network',
-               'options', 'comment']
+    _fields = ['ipv4addr', 'mac', 'network_view', 'extattrs', 'name',
+               'network', 'options', 'template', 'comment']
     _search_for_update_fields = ['ipv4addr', 'mac', 'network_view', 'network']
     _all_searchable_fields = _search_for_update_fields
     _shadow_fields = ['_ref', 'ip']
-    _return_fields = ['ipv4addr', 'mac', 'network_view', 'extattrs']
+    _return_fields = ['ipv4addr', 'mac', 'name', 'network_view', 'extattrs']
```
Comment (jkraj) [inline infoblox_client/objects.py]: return field doesn't need network?

Comment (ExaneServerTeam) [reply infoblox_client/objects.py]: @jkraj I'll check this point. Was not required on my side, but could be interesting to have this field available.

Comment (yuewko) [reply infoblox_client/objects.py]: My 2 cents...

Typically, you search/create a FixedAddress specifying `network` as a criteria - i.e. you already know the network - which I assume is why 'network' was never included in `_return_fields`.

Additionally, I believe `_return_fields` only specifies the *default* set of return fields. If needed, you can always add `network` to the set of return fields when doing a get/search. (see my comment here: https://github.com/infobloxopen/infoblox-client/pull/187#issuecomment-369620183)

File: infoblox_client/objects.py [hunk 2]
```diff
@@ -779,12 +779,13 @@ def ip(self, ip):
...
     _infoblox_type = 'fixedaddress'
-    _fields = ['ipv4addr', 'mac', 'network_view', 'extattrs', 'network',
-               'options', 'comment']
+    _fields = ['ipv4addr', 'mac', 'network_view', 'extattrs', 'name',
+               'network', 'options', 'template', 'comment']
     _search_for_update_fields = ['ipv4addr', 'mac', 'network_view', 'network']
     _all_searchable_fields = _search_for_update_fields
     _shadow_fields = ['_ref', 'ip']
-    _return_fields = ['ipv4addr', 'mac', 'network_view', 'extattrs']
+    _return_fields = ['ipv4addr', 'mac', 'name', 'network', 'network_view',
```
Comment (yuewko) [inline infoblox_client/objects.py]: As per my earlier comment [https://github.com/infobloxopen/infoblox-client/pull/188#discussion_r171590381](url), I'm not sure `network` belongs to the default set of return fields.

File: infoblox_client/objects.py [hunk 3]
```diff
@@ -800,11 +801,12 @@ def _build_options(members):
...
     """FixedAddress for IPv6"""
     _infoblox_type = 'ipv6fixedaddress'
-    _fields = ['ipv6addr', 'duid', 'network_view', 'extattrs', 'network',
-               'comment']
+    _fields = ['ipv6addr', 'duid', 'network_view', 'extattrs', 'name',
+               'network', 'template', 'comment']
     _search_for_update_fields = ['ipv6addr', 'duid', 'network_view', 'network']
     _all_searchable_fields = _search_for_update_fields
-    _return_fields = ['ipv6addr', 'duid', 'network_view', 'extattrs']
+    _return_fields = ['ipv6addr', 'duid', 'name', 'network', 'network_view',
```
Comment (yuewko) [inline infoblox_client/objects.py]: ditto

**Comment by saiprasannasastry** [comment]
@ExaneServerTeam  if you can remove the changes done for the _return fields i can merge this

**Comment by ExaneServerTeam** [comment]
I'll republish this merge request, as the code as moved on.

**Pull request closed without merging**